sdk/core/core/src/az_span_private.h

```diff
@@ -35,6 +35,64 @@ _az_span_scan_until(az_span self, _az_predicate predicate, int32_t* out_index);
 
 AZ_NODISCARD az_result _az_is_expected_span(az_span* self, az_span expected);
 
+/**
+ * @brief Removes all leading and trailing white space characters from the \p span. Function will create a new
+ * #az_span pointing to the first non-white-space (` `, \\n, \\r, \\t) character found in \p span and up to
+ * the last non-white-space character.
+ *
+ * @remarks If \p span is full of non-white-space characters, this function will return an empty #az_span slice.
+ *
+ * Example:
+ * \code{.c}
+ *  az_span a = AZ_SPAN_FROM_STR("  text with   \\n spaces   ");
+ *  az_span b = _az_span_trim_white_space(a);
+ *  // assert( b ==  AZ_SPAN_FROM_STR("text with   \\n spaces"));
+ * \endcode
+ *
+ * @param[in] source #az_span pointing to a memory address that might contain white spaces.
+ * @return The trimmed #az_span.
+ */
```

Comment:
Apply the same docs fixes you made here to the other methods as well.

- Use leading/trailing.
- Fix `@return` text.
- Fix `retrun` typo.
- `@remarks` should say return empty slice.
- etc.

Reply:
done